clients/repo_client.go

```diff
@@ -44,5 +44,6 @@ type RepoClient interface {
 	ListWebhooks() ([]Webhook, error)
 	ListProgrammingLanguages() ([]Language, error)
 	Search(request SearchRequest) (SearchResponse, error)
+	SearchCommits(request SearchCommitsOptions) (SearchCommitsResponse, error)
```

Comment:
IMO this API should return `[]Commit` so that this API is generic and can be easily extended in the future. I don't want to block this PR for that, so please leave a TODO so that one of us can pick it up later. 

If you can make the change either in this PR or in a future PR that'll be awesome and very much appreciated :)

Reply:
Updated SearchCommits and now returns `[]Commit` :+1: